src/app/pokemon-list/pokemon-list.component.ts

```diff
@@ -41,11 +41,9 @@ export class PokemonListComponent implements OnInit {
   pokemons!: PokemonWithImg[];
   pokemonsCount!: Observable<number>;
   destroyRef = inject(DestroyRef);
-  constructor(
-    private pokemonService: PokemonService,
-    public dialog: MatDialog,
-    private cdRef: ChangeDetectorRef
-  ) {}
+  private pokemonService = inject(PokemonService);
+  public dialog = inject(MatDialog);
```

Comment:
1) use public everywhere or don't use it at all;
2) use readonly everywhere is possible or don't use it at all

Reply:
ea59f22

Reply:
check again where to put readonly, or to be honest to not make your life very complex i would suggest to not to use readonly at all.

otherwise you may get even more comments from me :)

Reply:
I think we need to decide which approach to use in this project in relation to access modifiers. I just want to understand what are the best practices on modern real-world projects. I just met "the advice" (on YouTube) that it would be a good practice to specify 'private readonly' everywhere by default - and then change (expand access) as needed. Do I understand correctly that in this project I should not write readonly and public explicitly, but only private (if necessary)?

Reply:
you can decide on your own. real-world projects can vary. for one best practise is not best practise for others. 

here it really matters not approach, but consistency.

Reply:
private readonly everywhere is very strict approach, but if u are ready go ahead

Reply:
> private readonly everywhere is very strict approach, but if u are ready go ahead

I agree that this is an overly strict approach. just for a deeper understanding of this topic, I will try to follow this approach. rechecked access modifiers  ⬇️
963d1fa